src/excmds.ts

```diff
@@ -961,6 +974,183 @@ export function jumpprev(n = 1) {
+ * [a-z] are local marks, [A-Z] are global marks and '`' is the location before the last mark jump.
+ * @param key the key associated with the mark
+ */
+//#content
+export async function markjump(key: string) {
+    if (key.length !== 1) {
+        throw new Error("markjump accepts only a single letter or '`'")
+    }
+    if (key === "`") {
+        return markjumpbefore()
+    }
+    if (!/[a-z]/i.exec(key)) {
+        throw new Error("markjump accepts only a single letter or '`'")
+    }
+    if (key === key.toUpperCase()) {
+        return markjumpglobal(key)
+    }
+    return markjumplocal(key)
+}
+
+/**
+ * Jumps to a local mark.
+ * @param key the key associated with the mark
+ */
+//#content
+export async function markjumplocal(key: string) {
+    const urlWithoutAnchor = window.location.href.split("#")[0]
+    const localMarks = await State.getAsync("localMarks")
+    const mark = localMarks.get(urlWithoutAnchor)?.get(key)
+    if (mark) {
+        const currentTabId = await activeTabId()
+        state.beforeJumpMark = { url: urlWithoutAnchor, scrollX: window.scrollX, scrollY: window.scrollY, tabId: currentTabId}
+        scrolltab(currentTabId, mark.scrollX, mark.scrollY, `Jumped to mark '${key}'`)
+    }
+    return fillcmdline_tmp(3000, `Mark '${key}' is not set`)
+}
+
+/**
+ * Jumps to a global mark. If the tab with the mark no longer exists or its url differs from the mark's url,
+ * jumps to another tab with the mark's url or creates it first if such tab does not exist.
+ * @param key the key associated with the mark
+ */
+//#content
+export async function markjumpglobal(key: string) {
+    const globalMarks = await State.getAsync("globalMarks")
+    const mark = globalMarks.get(key)
+    if (!mark) {
+        return fillcmdline_tmp(3000, `Mark '${key}' is not set`)
+    }
+    const currentTabId = await activeTabId()
+    state.beforeJumpMark = {
+        url: window.location.href.split("#")[0],
+        scrollX: window.scrollX,
+        scrollY: window.scrollY,
+        tabId: currentTabId
+    }
+    try {
+        const tab = await browserBg.tabs.get(mark.tabId)
+        return onTabExists(tab)
+    } catch (e) {
+        return onTabNoLongerValid()
+    }
+
+    async function onTabExists(tab) {
+        const tabUrl = tab.url.split("#")[0]
+        if (mark.url !== tabUrl) {
+            return onTabNoLongerValid()
+        }
+        return goToTab(tab.id).then(() => {
+            scrolltab(tab.id, mark.scrollX, mark.scrollY, `Jumped to mark '${key}'`)
+        })
+    }
+
+    // the tab with mark's tabId doesn't exist or it has a different url than the mark's url
+    async function onTabNoLongerValid() {
+        const matchingTabs = await browserBg.tabs.query({url: mark.url})
+        // If there are no matching tabs, open a new one and update the mark's tabId for future use in this session
+        if (!matchingTabs.length) {
+            return openInNewTab(mark.url).then(updateMarkAndScroll())
```

Comment:
On my machine, this doesn't scroll to the right place once the tab is opened. We probably need to wait for the DocLoad event. I'll try to look into it today

Reply:
A broken sketch here:

```
            return openInNewTab(mark.url).then(tab => {
                mark.tabId = tab.id
                state.globalMarks = globalMarks
                browserBg.tabs.executeScript(
                    tab.id, {
                        code: `
                        window.addEventListener("DOMContentLoaded", () => window.scrollTo(${mark.scrollX}, ${mark.scrollY}))
                        `
                    }
                )
            })
```

I think "open tab and wait for it to load" is such a common operation that we should probably update `openInNewTab` to have an option that waits for the DOMContentLoaded event to fire before returning the promise.